### PR TITLE
Fixes and improvements for TensorRT

### PR DESCRIPTION
**Pull request by valeriosofi**
This PR brings the following changes:
- Fixes TensorRT behaviour when using dynamic shape. TensorRT requires to set a max, min and optimal value for each dynamic axis of the network inputs, so the dynamic_info structure has been changed in the following way:

**old_way**:

``` python
# Set dynamic info
dynamic_info = {
    "inputs": [
        {0: "batch", 2: "dim_image", 3: "dim_image"}
    ],
    "outputs": [
        {0: "batch", 1: "out_dim"}
    ]
}
```
**new_way**:

``` python
# Set dynamic info
dynamic_info = {
    "inputs": [
        {
            0: {"name": "batch", "min_val": 1, "opt_val": 1, "max_val": 4}, 
            2: {"name": "dim_image", "min_val": 224, "opt_val": 256, "max_val": 320},  
            3: {"name": "dim_image", "min_val": 224, "opt_val": 256, "max_val": 320},
        } 
    ],
    "outputs": [
        {0: "batch", 1: "out_dim"}
    ]
}
```
The old format is still supported, but does not work with TensorRT.

- Fixes some issues with unit tests, and adds additional controls to ensure that an optimized model with dynamic shape works properly.
- Improves significantly the performance of TensorRT with ONNX interface
- Removes all custom code for yolov5, because now can be optimized directly without wrappers, and updates the notebook
- Limits the gpu memory used by tensorflow, to avoid memory issues during tensorflow models optimization

### CODE REVIEW

File: apps/accelerate/speedster/docs/en/docs/advanced_options.md [hunk 1]
```diff
@@ -198,7 +203,26 @@ input_data = [((torch.randn(1, 3, 256, 256),), torch.tensor([0])) for _ in range
 # Set dynamic info
 dynamic_info = {
     "inputs": [
-        {0: "batch", 2: "dim_image", 3: "dim_image"}
+        {
+            0: {
+                "name": "batch",
+                "min_val": 1,
+                "opt_val": 1,
+                "min_val": 8,
```
Comment (diegofiori) [inline apps/accelerate/speedster/docs/en/docs/advanced_options.md:211]: shouldn't this be `max_val`?

File: nebullvm/operations/conversions/pytorch.py [hunk 2]
```diff
@@ -44,12 +44,20 @@ def convert_torch_to_onnx(
     input_names = [f"input_{i}" for i in range(len(input_tensors))]
     output_names = [f"output_{i}" for i in range(len(output_sizes))]
     dynamic_info = model_params.dynamic_info
+
     if dynamic_info is not None:
+        if isinstance(list(dynamic_info.inputs[0].values())[0], str):
+            onnx_format_inputs = dynamic_info.inputs
```
Comment (diegofiori) [inline nebullvm/operations/conversions/pytorch.py:50]: Add a comment explaining that this was done for backward compatibility reasons

File: nebullvm/operations/optimizations/compilers/quantizations/tensor_rt.py [hunk 3]
```diff
@@ -23,7 +22,7 @@ def quantize_tensorrt(
 ):
     if quantization_type is QuantizationType.HALF:
         config.set_flag(trt.BuilderFlag.FP16)
-        input_tfms.append(HalfPrecisionTransformation())
+        # input_tfms.append(HalfPrecisionTransformation())
```
Comment (diegofiori) [inline nebullvm/operations/optimizations/compilers/quantizations/tensor_rt.py:25]: if not needed anymore just delete the line (Maybe add a short comment about the reason why TensorRT does not need it)

File: apps/accelerate/speedster/docs/en/docs/advanced_options.md [hunk 4]
```diff
@@ -198,7 +203,26 @@ input_data = [((torch.randn(1, 3, 256, 256),), torch.tensor([0])) for _ in range
 # Set dynamic info
 dynamic_info = {
     "inputs": [
-        {0: "batch", 2: "dim_image", 3: "dim_image"}
+        {
+            0: {
+                "name": "batch",
+                "min_val": 1,
```
Comment (diegofiori) [inline apps/accelerate/speedster/docs/en/docs/advanced_options.md:209]: is it val or shape? In the code it seems to be shape, but in the docs you put val

Comment (valeriosofi) [reply apps/accelerate/speedster/docs/en/docs/advanced_options.md:209]: We use val to indicate the single axis dimension (each input has a shape, where each dimension has a min/max/opt val), while shape is the key used by tensor rt to compile the model with dynamic shape enabled. I called it differently because I didn't like using shape to refer to the single dimension of the shape.

**Review by diegofiori — APPROVED** [review]
LGTM!
